File: typescript/src/ExternalCustomer.ts
import {Address} from "@/Address";
import {ShoppingList} from "@/ShoppingList";

export class ExternalCustomer {
  constructor(public postalAddress: Address,
              public name: string,
              public preferredStore: string | null,
              public shoppingLists: ShoppingList[],
              public externalId: string,
              public companyNumber: string | null) {
  }

  isCompany() {
    return this.companyNumber !== null;
  }
}

File: java/src/main/java/codingdojo/model/ExternalCustomer.java
package codingdojo.model;

import lombok.Getter;
import lombok.Setter;
import lombok.experimental.Accessors;

import java.util.List;

@Getter
@Setter
@Accessors(chain = true)
public abstract class ExternalCustomer {
    private Address postalAddress;
    private String name;
    private String preferredStore;
    private List<ShoppingList> shoppingLists;
    private String externalId;

}
